File: Server/src/main/java/lk/wnl/wijeya/entity/PaperGrnPaper.java
package lk.wnl.wijeya.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "paper_grn_paper")
public class PaperGrnPaper {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "line_cost", precision = 10, scale = 2)
    private BigDecimal lineCost;

    @Column(name = "unit_price", precision = 7, scale = 2)
    private BigDecimal unitPrice;

    @Column(name = "quantity")
    private Integer quantity;
    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "paper_grn_id", nullable = false)
    private PaperGrn paperGrn;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "paper_id", nullable = false)
    private Paper paper;

}

File: Client/src/app/entity/PaperGrnPaper.ts
import {Paper} from "./Paper";

export class PaperGrnPaper {

  id!: number;
  paper!: Paper;
  quantity!: number;
  lineCost!: number;
  unitPrice!:number
  constructor(id: number, paper: Paper, quantity: number, lineCost: number, unitPrice: number) {
    this.id = id;
    this.paper = paper;
    this.quantity = quantity;
    this.lineCost = lineCost;
    this.unitPrice= unitPrice;
  }
}
